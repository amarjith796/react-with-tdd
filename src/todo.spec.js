import React from 'react';
import { shallow } from 'enzyme';
import TodoView from './Todo/todo';

describe("TodoView", () => {

	let wrapper;
	beforeEach(() => wrapper = shallow(<TodoView data={[]} handleDeleteSelected={jest.fn()} handleStatus={jest.fn()} handleDelete={jest.fn()} handleEditTodo={jest.fn()} />));
	it('should render correctly', () => expect(wrapper).toMatchSnapshot());
	it('should render a <div />', () => {
		expect(wrapper.find('div').length).toEqual(1);
	});
})