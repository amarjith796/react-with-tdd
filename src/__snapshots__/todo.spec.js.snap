// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoView should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodoView
    data={Array []}
    handleDelete={[MockFunction]}
    handleDeleteSelected={[MockFunction]}
    handleEditTodo={[MockFunction]}
    handleStatus={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          TODO TASK 
        </h2>,
        <button
          onClick={[Function]}
        >
          Delete Selected
        </button>,
        " ",
        <button>
          DeleteAll
        </button>,
        Array [],
      ],
      "style": Object {
        "background": "darkslategray",
        "minHeight": "600px",
        "width": "60vw",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "TODO TASK ",
          "style": Object {
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": "TODO TASK ",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Delete Selected",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Delete Selected",
        "type": "button",
      },
      " ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "DeleteAll",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "DeleteAll",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            TODO TASK 
          </h2>,
          <button
            onClick={[Function]}
          >
            Delete Selected
          </button>,
          " ",
          <button>
            DeleteAll
          </button>,
          Array [],
        ],
        "style": Object {
          "background": "darkslategray",
          "minHeight": "600px",
          "width": "60vw",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "TODO TASK ",
            "style": Object {
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": "TODO TASK ",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Delete Selected",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Delete Selected",
          "type": "button",
        },
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "DeleteAll",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "DeleteAll",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
